### PR TITLE
Update inject.js

### PR DESCRIPTION
**Pull request by mariusi7**
https://github.com/igrigorik/videospeed/issues/199

### CODE REVIEW

File: inject.js [hunk 1]
```diff
@@ -149,18 +149,14 @@ chrome.extension.sendMessage({}, function(response) {
...
-
-        case 'www.facebook.com':
-          // set stacking context to same as parent's parent.
+      if (location.hostname == 'www.facebook.com') {
+         // set stacking context to same as parent's parent.
           // + default fallthrough
           this.parent.style.zIndex = 'auto';
-
-        default:
+      } else if (location.hostname == 'www.amazon.com' || location.hostname.match('www\.hbogo\.*') {
```
Comment (igrigorik) [inline inject.js]: We should use test() instead of match(), as it returns true or false.

File: inject.js [hunk 2]
```diff
@@ -149,18 +149,14 @@ chrome.extension.sendMessage({}, function(response) {
       this.video.classList.add('vsc-initialized');
       this.video.dataset['vscid'] = this.id;
 
-      switch (location.hostname) {
```
Comment (igrigorik) [inline inject.js]: Technically, I think we can do regex matching in switch. Any particular reason why you prefer if's?

**Comment by mariusi7** [comment]
Didn't know JS can have expressions in switch, that's why I used "if" at first. I changed (needed to switch on 'true'). Also used "test()" instead of "match()". 
Note that I couldn't test hotdeploying the switch in debug mode no matter what I tried (either TypeError or thrown out of DevTools). So I  can only hope the syntax is correct.

**Comment by igrigorik** [comment]
> Note that I couldn't test hotdeploying the switch in debug mode no matter what I tried (either TypeError or thrown out of DevTools). So I can only hope the syntax is correct.

I installed the branch locally and everything seems to be working well. That said, I'm not sure I understand the issue above. What do you mean by "hot deploying"? Did you install the extension as an unpacked extension in Chrome? Or, are you just using the GitHub editor to change the code?

**Comment by mariusi7** [comment]
I only used the GitHub editor to change. Was unsure how to test, so I tried to copy-paste the changed switch while stopped in a breakpoint above it in Chrome.

**Comment by mariusi7** [comment]
I'll load it unpacked to make sure it actually works in my testcase...

**Comment by mariusi7** [comment]
Loaded the zip with my branch, disabled the original extension. All worked fine, video played, I could change the speed. So you can merge the changes from my POV if you agree too.

**Comment by mariusi7** [comment]
Unrelated to HBO not playing at all, I noticed now that the speed was reset a few times to 1.0 after I increase and let it play for ~10s. Or it starts at the preferred fast speed and resets to 1.0. This doesn't happen for local files or other video player. 
I read the other open/closed issues about reset.. Do you have any idea why? Buffering? 
It's not a VSC bug, it also resets on this player with it disabled and speed increased with another addon.

**Comment by igrigorik** [comment]
> I read the other open/closed issues about reset.. Do you have any idea why? Buffering?
> It's not a VSC bug, it also resets on this player with it disabled and speed increased with another addon.

This sounds like broken player logic within HBO player. Not much we can do about this..

Thanks for the patch, merging!

**Pull request merged**